### PR TITLE
Fix for #2956 

### PR DESCRIPTION
**Pull request by dahankzter**
Tried adding a new test case which seems to work.

Needs proper review since it is my first code on influxdb and I may very well violate some style or practise.

### CODE REVIEW

**Comment by dahankzter** [comment]
- [ ] CHANGELOG.md updated
- [ ] Rebased/mergable
- [ x] Tests pass
- [ x] Sign [CLA](http://influxdb.com/community/cla.html) (if not already signed)

Not sure what todo with point 2, taht state should be evident from the PR status right?
Ill update the changelog accordingly.

File: influxql/engine.go [hunk 1]
```diff
@@ -930,6 +932,16 @@ func (e *Executor) execute(out chan *Row) {
 	close(out)
 }
 
+func fromInt64ToFloat64(v interface{}) float64 {
```
Comment (jwilder) [inline influxql/engine.go]: Can you rename this to `i64tof64?  We usually keep their names succinct like: https://github.com/influxdb/influxdb/blob/master/tsdb/shard.go#L806

File: influxql/engine.go [hunk 2]
```diff
@@ -930,6 +932,16 @@ func (e *Executor) execute(out chan *Row) {
...
+	switch v.(type) {
+	case int64:
+		return float64(v.(int64))
+	case float64:
+		return v.(float64)
+	}
+	panic(fmt.Sprintf("Expected either int64 or float64, got %v", v))
```
Comment (jwilder) [inline influxql/engine.go]: The should be all lower-case.

**Comment by jwilder** [comment]
Two small things but looks good otherwise.

**Comment by dahankzter** [comment]
Did I understand the last bit about case? I interpreted it as the actual error line should be lower case.

File: influxql/engine.go [hunk 3]
```diff
@@ -408,7 +409,8 @@ func (m *MapReduceJob) processRawQueryDerivative(lastValueFromPreviousChunk *raw
 
 		// Calculate the derivative of successive points by dividing the difference
 		// of each value by the elapsed time normalized to the interval
-		diff := v.Values.(float64) - lastValueFromPreviousChunk.Values.(float64)
+		diff := i64tof64(v.Values) - i64tof64(lastValueFromPreviousChunk.Values)
```
Comment (otoolep) [inline influxql/engine.go]: It seems to me to be more idiomatic to spell this out. `int64toFloat64`. Look at the math package.

**Comment by otoolep** [comment]
I'd really like to see the function names fully spelled out.

**Comment by dahankzter** [comment]
Yes I guess Go's math package has funcs like Float32frombits for example. That makes sense but whats more important perhaps is what the coding style Influxdb has chosen. The Scanner struct in scanner.go has a number of helper funcs like "isIdentFirstChar", "isLetter" and ast.go has funcs like "IsSimpleDerivative".

What do you say @jwilder @otoolep perhaps the longer form is more Influxdb idiomatic?

Personally I have no real preference, I have both seen and used both in other projects/languages but consistency is quite important so if there is an preferred Influxdb way we should adhere to it.

**Comment by dahankzter** [comment]
Doh, I merged upstream master...

Ill try to revert it properly to create a clean PR.

Any eta and more comments about the PR per se? @jwilder @otoolep

**Comment by otoolep** [comment]
Spell out the function names, that is my only request @dahankzter

**Comment by dahankzter** [comment]
Ok, ill clean up the PR and ping you here @otoolep

**Comment by gunnaraasen** [comment]
@dahankzter The 0.9.1 release branch for testing will be cut soon. It would be nice if this fix made it in. Any chance you can clean up this PR today?

**Comment by jwilder** [comment]
Manually merged this https://github.com/influxdb/influxdb/commit/4d2915b49e5e4a41ec70f3fb3d32bb5500d05df2

**Comment by dahankzter** [comment]
Sorry guys I have been really busy. Ill make a new pull request with the last rename to the longer format but I guess that is not such a hurry since it has no bearing on the actual issue.

Thank you @jwilder and @otoolep for your patience!

**Pull request closed without merging**